commands/create_cluster_test.go

```diff
@@ -76,3 +96,47 @@ func Test_CreateFromBadYAML01(t *testing.T) {
 		t.Error("Expected owner to be empty, got: ", myDef.Owner)
 	}
 }
+
+// Test_CreateFromCommandLine tests a cluster creation completely based on
+// command line arguments
+func Test_CreateFromCommandLine(t *testing.T) {
+	// mock server always responding positively
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Printf("mockServer request: %s %s\n", r.Method, r.URL)
+		w.Header().Set("Content-Type", "application/json")
+		w.Header().Set("Location", "/v4/clusters/f6e8r/")
+		w.WriteHeader(http.StatusCreated)
+		w.Write([]byte(`{"code": "RESOURCE_CREATED", "message": "Yeah!"}`))
+
+	}))
+	defer mockServer.Close()
+
+	// first run, with minimal arguments
+	args := []string{}
+	cmdAPIEndpoint = mockServer.URL
+	cmdOwner = "acme"
+	checkAddCluster(CreateClusterCommand, args)
+	addCluster(CreateClusterCommand, args)
+
+	// second run, with additional arguments
+	cmdClusterName = "UnitTestCluster"
```

Comment:
I see.

The `args` parameter isn't used at all in the tests. It's just a parameter that's always added when cobra executes the command. I'm not sure, but I think it would be used by cobra if there were additional arguments which aren't mapped to any of the named or positional arguments we have defined.

See https://github.com/spf13/cobra/blob/master/command.go#L90-L92

So basically I'm setting parameters via the `cmd*` variables (like `cmdVerbose`), which are normally set by the user's command line arguments. Then I'm executing the functions  `checkAddCluster` (for validation) and `addCluster`(for final execution).